### PR TITLE
Autojoin furyagent command

### PR DESCRIPTION
**Pull request by lzecca78**
to be more complete and resilient, furyagent now will be owner of the join process of the node.
I decided to expose an additional parameter to the `furyagent.yml` config file: 

```yaml
........
clusterComponent:
  node:
    joinCmdMaxRetryMin: 20
```
that identify the max window time for the exponential backoff to retry. For the exponential backoff i used this library : https://godoc.org/gopkg.in/cenkalti/backoff.v4 that implements a non linear exponential backoff (see tech details here : https://godoc.org/gopkg.in/cenkalti/backoff.v4#ExponentialBackOff) .
Basically is a really simple implementation: the download of the join script from s3 bucket and the execution of it is wrapped around the backoff logic. This to avoid every race condition/ network partitioning that may occur during the very first execution after the boot process.
Is still to be tested, i would like to do this in pair with @angelbarrera92 to become friendly on the flow and put this under ci possibly in order to avoid regressions in the future.
Let me know if you some inconsistencies or enhancements that could be done!

### CODE REVIEW

**Comment by lzecca78** [comment]
[Auto-Join bash logic](https://trello.com/c/BxmiIhGe/22-auto-join-bash-logic)

**Comment by angelbarrera92** [comment]
LGTM! Pending pair testing ;) but awesome work Luca ;)

File: pkg/component/component.go [hunk 1]
```diff
@@ -82,7 +83,8 @@ type MasterConfig struct {
 
 // NodeConfig is used to backup/restore/configure worker nodes (backup and restore have an empty implementation right now)
 type NodeConfig struct {
-	CloudProvider string `mapstructure:"caKeyFilename"`
+	CloudProvider string        `mapstructure:"caKeyFilename"`
+	RetryMaxMin   time.Duration `mapstructure:""joinCmdMaxRetryMin`
```
Comment (lnovara) [inline pkg/component/component.go]: Can we find a better name for the config option?

In addition, do you think it could make sense to provide a reasonable default?

Comment (lzecca78) [reply pkg/component/component.go]: I know the name of variable is one of the hardest thing to do with the cache invalidation. Please if you have a better one, please feel free to suggest! About the default value, usually I am against defaults. But if you all are agree on a default, we can set one!

Comment (lnovara) [reply pkg/component/component.go]: Could `joinTimeout` be a good name?

About the default, as I think we'll always end up using the same value everywhere, why not choose a reasonable default?

What do you think?

cc @phisco @ralgozino @angelbarrera92

Comment (lzecca78) [reply pkg/component/component.go]: 30min could be something acceptable? Or more?

Comment (lzecca78) [reply pkg/component/component.go]: btw @lnovara  `joinTimeout` is surely a better name :D

Comment (lnovara) [reply pkg/component/component.go]: Yes, 30 minutes should be an acceptable default from my POV.

Comment (ralgozino) [reply pkg/component/component.go]: I agree. 30 minutes is a sane default and `joinTimeout` is much better :)

Comment (lzecca78) [reply pkg/component/component.go]: perfect. i just made both the modifications:
1. renaming the awful var type in the struct
2. adding the default logic for joinTimeout

Let me know. @angelbarrera92  when you want, we can test, if you have time, otherwise let's schedule a 30min timeframe to do that togheter 👍

**Comment by jnardiello** [comment]
Assigned to @angelbarrera92 for final merge.

**Comment by angelbarrera92** [comment]
We've got schedule a demo about this feature for today in the morning ;) after that, we can start thinking in merge this feature ;)

**Comment by angelbarrera92** [comment]
Tested, and merging! Thanks @lzecca78!

**Comment by jnardiello** [comment]
Thanks everyone
